shell/browser/api/electron_api_web_contents.cc

```diff
@@ -2514,9 +2537,9 @@ void WebContents::Print(gin::Arguments* args) {
     settings.SetIntKey(printing::kSettingDpiVertical, dpi);
   }
 
-  base::ThreadPool::PostTaskAndReplyWithResult(
-      FROM_HERE, {base::MayBlock(), base::TaskPriority::USER_BLOCKING},
-      base::BindOnce(&GetDefaultPrinterAsync),
+  scoped_refptr<base::TaskRunner> tr = CreatePrinterHandlerTaskRunner();
+  base::PostTaskAndReplyWithResult(
+      tr.get(), FROM_HERE, base::BindOnce(&GetDefaultPrinterAsync),
```

Comment:
* The task runner will not be available after this call, since the `base::PostTaskAndReplyWithResult` takes it in as raw pointer. We should store the task runner as a member.

* The call here can instead be

```
print_task_runner_->PostTaskAndReplyWithResult()
```

Reply:
@deepak1556 i think we do need to use `base::PostTaskAndReplyWithResult` like [upstream](https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/ui/webui/print_preview/local_printer_handler_default.cc;l=136-140;drc=8e5afb8dfb07d9f6f6f7fa5ad65c57cb5927097e;bpv=1;bpt=1)